### PR TITLE
PulseAudio Volume Control (pavucontrol)

### PR DESCRIPTION
**Pull request by wjt**
Endless OS users, myself included, sometimes wish this app were available; spurred on by a discussion on a Telegram group, I threw together a manifest.

TODO:

* [x] Remove many libcanberra files
* [x] Update *mm dependencies (ideally adding them to shared-modules)
* [x] Rewrite manifest to YAML, with consistent layout
* [x] Explain --disable-oss (libcanberra's configure prints a SHOUTY
      WARNING if you leave it enabled)
* [x] Add appdata
* [x] Add screenshots
* [x] Fix icon situation:
  - WARNING: Icon not matching app id in /sysroot/home/wjt/src/flathub/flathub/_build/export/share/applications/org.pulseaudio.pavucontrol.desktop: multimedia-volume-control
  - WARNING: Icon referenced in desktop file but not exported: multimedia-volume-control

Deferred:

* [ ] Upstream appdata
  - Not entirely trivial because upstream uses intltool; to do this properly and have it wired up for translations, we'll want to switch it to using xgettext.
  - Let's pursue this once the appdata is confirmed to be Good Enough For Flathub

### CODE REVIEW

**Comment by wjt** [comment]
@ford-prefect FYI. Would anyone in the PulseAudio team be interested in maintaining this (ie, updating pavucontrol and its dependencies when a new release is made) if I finish this?

**Comment by nedrichards** [comment]
bot, build org.pulseaudio.pavucontrol

**Comment by ford-prefect** [comment]
@wjt I'm happy to help shepherd this, and can drop a note on the PulseAudio list once we have it in place to get some more volunteers.

**Comment by nedrichards** [comment]
bot, build org.pulseaudio.pavucontrol

**Comment by btreut** [comment]
I successfully tried pavucontrol on Q4OS (Debian Stretch based distro) and EndlessOS and it seems to work properly. But I probably scratched only the surface of the what pavucontrol provides, so use my results with caution.

**Comment by nedrichards** [comment]
bot, build org.pulseaudio.pavucontrol

**Comment by ford-prefect** [comment]
I've uploaded some boring screenshots based on GNOME in Fedora 30:

  https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-playback.png
  https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-record.png
  https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-output.png
  https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-input.png
  https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-config.png

File: org.pulseaudio.pavucontrol.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        </ul>
+      </description>
+    </release>
+  </releases>
+  <content_rating type="oars-1.1" />
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
+  <url type="homepage">https://freedesktop.org/software/pulseaudio/pavucontrol/</url>
+  <url type="bugtracker">https://gitlab.freedesktop.org/pulseaudio/pavucontrol/issues</url>
+  <updatecontact>https://gitlab.freedesktop.org/pulseaudio/pavucontrol/issues/new</updatecontact>
```
Comment (bilelmoussaoui) [inline org.pulseaudio.pavucontrol.appdata.xml]:
```suggestion
  <update_contact>https://gitlab.freedesktop.org/pulseaudio/pavucontrol/issues/new</update_contact>
```

**Comment by wjt** [comment]
Need to still fix:

```

**Comment by ford-prefect** [comment]
@wjt we've dropped the `intltool` dependency in git fwiw, so next release won't need it.

**Comment by wjt** [comment]
> @wjt we've dropped the `intltool` dependency in git fwiw, so next release won't need it.

I don't see this change in pavucontrol master https://gitlab.freedesktop.org/pulseaudio/pavucontrol/blob/master/configure.ac#L67 – am I looking in the wrong place?

**Comment by wjt** [comment]
bot, build org.pulseaudio.pavucontrol

**Comment by wjt** [comment]
Warnings from strict validation:

```
• value-missing         : <!-- Copyright [year] [name] --> is not present
• attribute-invalid     : <screenshot> width too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-config.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-config.png] maximum is 900px
• aspect-ratio-invalid  : <screenshot> aspect ratio not 16:9 [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-config.png]
• attribute-invalid     : <screenshot> width too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-input.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-input.png] maximum is 900px
• aspect-ratio-invalid  : <screenshot> aspect ratio not 16:9 [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-input.png]
• style-invalid         : <caption> is too long [The “Input” tab, showing connected input devices];longest allowed is 50 chars
• attribute-invalid     : <screenshot> width too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-output.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-output.png] maximum is 900px
• aspect-ratio-invalid  : <screenshot> aspect ratio not 16:9 [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-output.png]
• style-invalid         : <caption> is too long [The “Output” tab, showing connected output devices];longest allowed is 50 chars
• attribute-invalid     : <screenshot> width too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-playback.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-playback.png] maximum is 900px
• aspect-ratio-invalid  : <screenshot> aspect ratio not 16:9 [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-playback.png]
• style-invalid         : <caption> is too long [The “Playback” tab, showing applications currently playing audio];longest allowed is 50 chars
• attribute-invalid     : <screenshot> width too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-record.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-record.png] maximum is 900px
• aspect-ratio-invalid  : <screenshot> aspect ratio not 16:9 [https://freedesktop.org/software/pulseaudio/pavucontrol/screenshots/pavucontrol-record.png]
• style-invalid         : <caption> is too long [The “Recording” tab, showing applications currently recording audio];longest allowed is 50 chars
• style-invalid         : <li> cannot end in '.' [Improve the use of space (remove useless margins and paddings).]
• style-invalid         : <li> cannot end in '.' [Use a more appropriate icon for the channel lock button.]
• style-invalid         : <li> cannot end in '.' [Better channel label layout, prevents volume sliders from getting unaligned.]
• style-invalid         : <li> is too long [Maximum latency offset increased from 2 to 5 seconds to accommodate AirPlay devices that often have higher latency than 2 seconds (this is not that useful on newer PulseAudio versions, though, because the latency is reported much more accurately than before).] maximum is 250 chars
• style-invalid         : <li> cannot end in '.' [Maximum latency offset increased from 2 to 5 seconds to accommodate AirPlay devices that often have higher latency than 2 seconds (this is not that useful on newer PulseAudio versions, though, because the latency is reported much more accurately than before).]
• style-invalid         : <li> cannot end in '.' [New --version command line option.]
• style-invalid         : <li> cannot end in '.' [New translations: Chinese (Taiwan), Croatian, Korean, Norwegian Nynorsk, Lithuanian, Valencian.]
• style-invalid         : <li> cannot end in '.' [Updated translations: Finnish, French, German, Italian, Japanese, Polish, Swedish.]
• style-invalid         : Not enough <p> tags for a good description [0/1]
• style-invalid         : <summary> is shorter than <name>
• translations-required  : <name> has no translations
• translations-required  : <summary> has no translations
• translations-required  : <description> has no translations
```

**Comment by wjt** [comment]
I've reduced the length of the captions (I wasn't sure about them anyway) and made some other tweaks, but I'm not going to change the release notes – they are taken verbatim from https://freedesktop.org/software/pulseaudio/pavucontrol/.

@ford-prefect I can't bring myself to ask you to take low-dpi 16:9 screenshots… :)

The translations would come by integrating the appdata into the upstream project.

**Comment by wjt** [comment]
bot, build org.pulseaudio.pavucontrol

**Comment by nedrichards** [comment]
@wjt fwiw a number of us in flathub disagree with the strict-validation rules, which is why they're a warning rather than a fail. It looks like most of the WIP stuff has been resolved, is this ready for potential inclusion?

**Comment by wjt** [comment]
Yes please!

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.pulseaudio.pavucontrol

**Comment by nedrichards** [comment]
let us know if any other pulse-ists need to be added to the repo.

**Comment by ford-prefect** [comment]
@wjt I've updated the screenshots to be 1600x900 now. If that's too large, do let me know and I can halve them.

I was wrong about pavucontrol dropping intltool, that was just the main PulseAudio repo.

**Comment by ford-prefect** [comment]
jfyi, tracking moving appstream data upstream at: https://github.com/flathub/org.pulseaudio.pavucontrol/issues/3

**Pull request closed without merging**